### PR TITLE
Add platform and libraries documentation

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes changes to root README related to filling out the Platforms and Libraries section:

- libraries and platforms used in production code
- libraries used in test code

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -258,6 +258,26 @@ We do not repeat the information covered in the previous chapter, we just extend
...
+- https://github.com/dotnet/efcore[Entity Framework]
+- https://github.com/npgsql/npgsql[Npgsql]
+
+Testing:
+
+- https://github.com/xunit/xunit[xunit]
+- https://github.com/moq/moq[Moq]
+- https://github.com/VerifyTests/Verify[Verify]
+- https://github.com/bchavez/Bogus[Bogus]
+- https://github.com/fluentassertions/fluentassertions[FluentAssertions]
```
Comment (kamilbaczek) [inline README.adoc:281]:
```suggestion
- https://github.com/fluentassertions/fluentassertions[FluentAssertions]
- https://dotnet.testcontainers.org[Testcontainers]
```

File: README.adoc [hunk 2]
```diff
@@ -258,6 +258,27 @@ We do not repeat the information covered in the previous chapter, we just extend
 
 IMPORTANT: If you want to get the most out of this repository, we recommend that you read the first chapter, understand the code and description, and then navigate to another folder. Of course you can go through the folders in your own way but you might miss some concepts.
 
-== Technologies and Tools Used
+## Libraries and Platforms Used
```
Comment (kamilbaczek) [inline README.adoc:261]: I think we should change ## to == like in other part of file. Are you okay?
@meaboutsoftware

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
I think we should change ## to == like in other part of file. Are you okay?

File: README.adoc (hunk 2 above)
Comment (meaboutsoftware) [reply README.adoc]: Done

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**